### PR TITLE
Pr #132 integration

### PR DESCRIPTION
**Pull request by c0c0n3**
This PR integrates the fixes to #127 and #129 that @agaldemas kindly contributed.

Points to note:

1. PR #132 is the initial contribution from @agaldemas: a fix for #127, #128 and #129.
2. Since PR #136 already addresses #128, this PR only integrates the code @agaldemas contributed for #127 and #129 while discards contributions towards fixing #128---for the rationale behind this, refer to PR #132.
3. This PR should be merged into `v0.6rc` **after** merging #132.

### CODE REVIEW

File: src/translators/crate.py [hunk 1]
```diff
@@ -284,6 +288,22 @@ def _insert_entities_of_type(self,
...
         return self.cursor
 
+    def _attr_is_structured(self, a):
+        # isinstance(a, object)
+        isdict = False
+
+        if a['value'] is not None and isinstance(a['value'], dict):
+            self.logger.info("attribute {} has 'value' attribute of type dict".format(a))
+            isdict = True
+        # the next test is probably unuseful
```
[the remark this answers is not in the document]
Comment (chicco785) [reply src/translators/crate.py]: i think we should not store it.

File: src/translators/crate.py [hunk 2]
```diff
@@ -234,14 +234,18 @@ def _insert_entities_of_type(self,
...
 
                 if attr_t not in NGSI_TO_CRATE:
-                    supported_types = ', '.join(NGSI_TO_CRATE.keys())
-                    msg = ("'{}' is not a supported NGSI type. "
-                           "Please use any of the following: {}. "
-                           "Falling back to {}.")
-                    self.logger.warning(msg.format(
-                        attr_t, supported_types, NGSI_TEXT))
+                    # if attribute is complex assume it as an NGSI StructuredValue
+                    if self._attr_is_structured(e[attr]):
```
[the remark this answers is not in the document]
Comment (chicco785) [reply src/translators/crate.py]: it is a missing feature... at the time being, we support only array of strings (which I agree is limited)

File: specification/quantumleap.yml [hunk 3]
```diff
@@ -125,14 +125,14 @@ definitions:
...
-    name: fiwareService
+    name: fiware-Service
     required: false
     type: string
     description: "The corresponding 'fiware-service' header.
     See http://fiware-orion.readthedocs.io/en/latest/user/multitenancy/index.html#multi-tenancy"
   fiwareServicePath:
     in: header
-    name: fiwareServicePath
+    name: fiware-ServicePath
```
[the remark this answers is not in the document]
Comment (chicco785) [reply specification/quantumleap.yml]: probably yes, but he i suspect he is right, it was a bug.

File: specification/quantumleap.yml [hunk 4]
```diff
@@ -125,14 +125,14 @@ definitions:
 parameters:
   fiwareService:
     in: header
-    name: fiwareService
+    name: fiware-Service
```
[the remark this answers is not in the document]
Comment (chicco785) [reply specification/quantumleap.yml]: probably yes, but i suspect he is right, it was a bug.

File: src/translators/crate.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (c0c0n3) [reply src/translators/crate.py]: I think it gets stored with a null value because of #125 , but should double check. 
I think if we've seen that attribute before, we should store it with a null value---again see discussion in #125. But I agree with you in the case we haven't seen that attribute yet. In other words, if an attribute A comes in with no value nor type---e.g. `A = { k: 123, value: null, type: ''}`---and this is the *first* time we save the enclosing entity, then we should exclude A from the list of attrs we stash away as opposed to creating a column A of type `string`. This way, if the next time around the entity does have an attribute A with a type which isn't text, we can still store it without problems....

File: src/translators/crate.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (c0c0n3) [reply src/translators/crate.py]: Okie, but should we implement it as part of this PR? The reason why I'm saying this is that if we want to support dictionaries then we should do it consistently across the board, to avoid clients getting confused about when they can use a dictionary...

[the remark this answers is not in the document]
Comment (c0c0n3) [reply src/translators/crate.py]: opened a separate issue (#147) for this to avoid putting too many things in this PR :-)

File: specification/quantumleap.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (c0c0n3) [reply specification/quantumleap.yml]: rolled back the change and opened an issue for this---see #148

File: src/translators/crate.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (c0c0n3) [reply src/translators/crate.py]: Zapped the comment, added a test case for it (`test_store_empty_attribute_as_null_text`) and opened an issue to review translation corner cases: #152.

**Comment by c0c0n3** [comment]
Closes #127, #129.
